modules/library/main/src/main/java/org/geotools/util/ColorConverterFactory.java

```diff
@@ -43,66 +43,92 @@
  */
 public class ColorConverterFactory implements ConverterFactory {
 
+    /**
+     * Uses {@link Color#decode(String)} to convert String to Color.
+     */
+    public static Converter CONVERT_STRING = new Converter() {
+        public <T> T convert(Object source, Class<T> target) throws Exception {
+            String rgba = (String) source;
+            try {
+                return target.cast(Color.decode(rgba));
+            } catch (NumberFormatException badRGB) {
+                // unavailable
+                return null;
+            }
+        }
+    };
+
+    /**
+     * Converts provided integer to color, taking care to allow rgb and rgba support.
+     */
+    public static Converter CONVERT_NUMBER = new Converter() {
+        public <T> T convert(Object source, Class<T> target) throws Exception {
+            Number number = (Number) source;
+            // is it an integral number, and small enough to be an integer?
+            if (((int) number.doubleValue()) == number.doubleValue()
+                    && number.doubleValue() < Integer.MAX_VALUE) {
+                int rgba = number.intValue();
+                int alpha = 0xff000000 & rgba;
+                return target.cast(new Color(rgba, alpha != 0));
+            } else {
+                return null;
+            }
+        }
+    };
+
+    /**
+     * Converts color to hex representation.
+     */
+    private Converter CONVERT_COLOR_TO_STRING = new Converter() {
+
+        public <T> T convert(Object source, Class<T> target) throws Exception {
+            Color color = (Color) source;
+
+            String redCode = Integer.toHexString(color.getRed());
```

Comment:
I'm not sure it's strictly necessary -- as it is now, when trying to convert a string to a color, this is what happens:

(1) If you provide `Hints.COLOR_NAMES="CSS"`: You get a converter that tries the color map and falls back to the ordinary `CONVERT_STRING` converter.
(2) If you don't provide the hint: You just get back the `CONVERT_STRING` converter.

An additional check in the factory would just duplicate the fallback logic in the css converter. Of course, I wouldn't mind it, if you want to be extra safe.